services/brig/src/Brig/Run.hs

```diff
@@ -156,10 +160,15 @@ type ServantCombinedAPI =
       :<|> Servant.Raw
   )
 
-lookupRequestIdMiddleware :: (RequestId -> Wai.Application) -> Wai.Application
-lookupRequestIdMiddleware mkapp req cont = do
-  let reqid = maybe def RequestId $ lookupRequestId req
-  mkapp reqid req cont
+lookupRequestIdMiddleware :: Logger -> (RequestId -> Wai.Application) -> Wai.Application
+lookupRequestIdMiddleware logger mkapp req cont = do
+  case lookupRequestId req of
+    Just rid -> do
+      mkapp (RequestId rid) req cont
+    Nothing -> do
+      localRid <- RequestId . cs . UUID.toText <$> UUID.nextRandom
+      Log.info logger $ "request-id" .= localRid ~~ "request" .= (show req) ~~ msg (val "generated a new request id for local request")
```

Comment:
I think it would be nicer if log basics of a request like method and path instead of logging the whole request. It could contain sensitive info in headers. This way the logs also become searchable by those things.